### PR TITLE
Add a `--all` flag for `check` subcommand

### LINKED ISSUE #398: Run `check` on all projects if no project is specified
I am writing a pre-push hook that builds and tests all projects. The `moon check` is the perfect command for this. However I have to list all the projects projects manually. This is error prone and I have to remember to update the command everytime I add a project.

I propose a `--all` flag to the `moon check` command that will run the command on all the projects configured in the `.moon/workspace.yml` file.

### PR DESCRIPTION
**Pull request by IgnisDa**
Fixes #398.

### CODE REVIEW

**Comment by milesj** [comment]
@IgnisDa I think an `--all` option would be better, otherwise this wouldn't work when not ran from the root.

**Comment by IgnisDa** [comment]
@milesj I did not think of that possibility. I added a `--all` flag and added 2 tests for it. Please review.

File: crates/cli/src/commands/check.rs [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
 
 pub struct CheckOptions {
     pub report: bool,
 }
 
+const LOG_TARGET: &str = "moon:check";
+
 pub async fn check(
     project_ids: &Vec<String>,
+    all: bool,
```
Comment (milesj) [inline crates/cli/src/commands/check.rs:15]: can you put the all in `CheckOptions`

File: crates/cli/src/app.rs [hunk 2]
```diff
@@ -289,10 +289,15 @@ pub enum Commands {
     )]
     Check {
         #[arg(help = "List of project IDs to explicitly check")]
+        #[clap(group = "projects")]
```
Comment (milesj) [inline crates/cli/src/app.rs:292]: what does this do exactly?

Comment (IgnisDa) [reply crates/cli/src/app.rs:292]: This makes `[ID]s` and `--all` mutually exclusive so that they can not be used together.

Comment (IgnisDa) [reply crates/cli/src/app.rs:292]: image

Comment (milesj) [reply crates/cli/src/app.rs:292]: Awesome

**Pull request merged**